xarray/tests/test_backends.py

```diff
@@ -1895,20 +1895,23 @@ def test_chunk_encoding_with_dask(self):
             # don't actually check equality because the data could be corrupted
             pass
 
+        # if dask chunks (4) are an integer multiple of zarr chunks (2) it should not fail...
```

Comment:
This is a good catch! You're absolutely right that this should work. Perhaps you want to rename the variable below to something other than `badenc`, since it is not actually bad...